ports/raspberrypi/common-hal/mdns/Server.c

```diff
@@ -337,7 +330,15 @@ void common_hal_mdns_server_advertise_service(mdns_server_obj_t *self, const cha
     if (existing_slot < MDNS_MAX_SERVICES) {
         mdns_resp_del_service(NETIF_STA, existing_slot);
     }
-    int8_t slot = mdns_resp_add_service(NETIF_STA, self->instance_name, service_type, proto, port, srv_txt, (void *)txt_records);
+
+    srv_txt_t *txt_userdata = malloc(sizeof(srv_txt_t)); // TODO: Free this eventually
+    txt_userdata->num_txt_records = num_txt_records;
+    txt_userdata->txt_records = malloc(sizeof(char *) * num_txt_records);
+    for (size_t i = 0; i < num_txt_records; i++) {
+        txt_userdata->txt_records[i] = strdup(txt_records[i]);
+    }
```

Comment:
Q: do these strings need to live past the call to `mdns_resp_add_service()`, or could you allocate them on the stack?

Reply:
These do need to live for longer since `txt_userdata` is used by the `srv_txt` callback within lwIP to respond to requests for TXT records. It may be called at any time and multiple times after starting the mdns service.